### DISCUSSION TITLE
Is this the best way to map the members of a class to keys for multiple objects?

### DISCUSSION THREAD

**Discussion by dominique120** [nlohmann/json#2388] [Q&A] [answered]
So I have a pretty standard class called Client with its usual members(id, name, etc) and I've succeed in getting it mapped to its keys for each instance of the object and I have the output I want. 

However I'm not convinced I'm doing it the right way. Is there a better way to do this?

This is the code I wrote:
```
	Client client;
	std::vector<Client> cl = client.selectClients();
	nlohmann::json j;

	for (int i = 0; i < cl.size(); ++i) {
		j.push_back( {
			{ "ClientId",cl.at(i).ClientId},
			{ "FirstName", cl.at(i).FirstName},
			{ "LastNames", cl.at(i).LastNames},
			{ "PrimaryAddress", cl.at(i).PrimaryAddress},
			{ "SecondaryAddress", cl.at(i).SecondaryAddress},
			{ "PrimaryPhone", cl.at(i).PrimaryPhone},
			{ "SecondaryPhone", cl.at(i).SecondaryPhone},
			{ "District", cl.at(i).District}
			});
	}
```

The format I get is as follows and its exactly as I want it:
```
{
   {
   "id" : 1,
   "name" : "john",
   "lastname: "doe"
   },
   {
   "id" : 2,
   "name" : "jane",
   "lastname: "doe"
   },
}

Thanks!

**Comment by nlohmann** [comment, marked as answer]
You may want to define a `to_json` function for your type `Client`. See https://github.com/nlohmann/json#arbitrary-types-conversions for more information.

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?
